engine/src/main/java/io/camunda/zeebe/engine/processing/bpmn/behavior/BpmnJobBehavior.java

```diff
@@ -158,6 +164,34 @@ private Either<Failure, String> evalDateExp(final Expression date, final long sc
         .map(optionalDate -> optionalDate.map(ZonedDateTime::toString).orElse(null));
   }
 
+  private Either<Failure, String> evalFormIdExp(final Expression formIdExp, final long scopeKey) {
+    if (formIdExp == null) {
+      return Either.right(null);
+    }
+    return expressionBehavior
+        .evaluateStringExpression(formIdExp, scopeKey)
+        .flatMap(
+            formId -> {
+              final Optional<PersistedForm> latestFormById =
+                  formState.findLatestFormById(wrapString(formId));
+              return latestFormById
+                  .<Either<Failure, String>>map(
+                      persistedForm -> Either.right(String.valueOf(persistedForm.getFormKey())))
+                  .orElseGet(
+                      () ->
+                          Either.left(
+                              new Failure(
+                                  String.format(
+                                      "Expected to find a form with id '%s',"
+                                          + " but no form with this id it is found,"
```

Comment:
@nicpuppa A small typo:
```suggestion
                                          + " but no form with this id is found,"
```